sphinx/environment/__init__.py

```diff
@@ -81,9 +81,7 @@
 
 if TYPE_CHECKING:
     from collections.abc import MutableMapping
-    from typing import Literal
-
-    from typing_extensions import overload
+    from typing import Literal, overload
```

Comment:
I don't get how this was working _now_. Nothing imports `typing-extensions` into the project, that I can see. It's not listed in any of the dependencies lists, and a `pip install -e .[test]` from the project root doesn't pull it into the current venv, even when running Python 3.9.

Even more confusing, it _never has_ been imported into the project, AFAICT. a `git log -G 'typing.extensions' turns up nothing but two edits to **this file** back in 2022.

Reply:
Does Mypy automatically import it as needed, or something?

Reply:
Never mind, `mypy` does indeed have a dependency on `typing_extensions`. I hadn't realized _Mypy_ wasn't part of the `sphinx[test]` dependencies either.

Reply:
Well spotted - I scanned that line a few times but didn't think to check the origin of the module.  I suppose it's somewhat-hidden by the fact that the `TYPE_CHECKING` paths are evaluated under different conditions (such as when using `mypy` as you mention).

FWIW: That does appear to be the _only_ reference to `typing_extensions` in the source.  There are a few non-`sphinx`-internal imports within `TYPE_CHECKING` guards; mostly to `docutils` and `jinja` (relatively safe, stable dependencies), and some Python internal deps (again OK until such time as those interfaces change).

Reply:
\*nod\* I noticed that it was the only instance of `typing_extensions` in the codebase, and (with these changes) the only surviving import from it is the `@overload` decorator.

Which actually existed in Python 3.9, even, so it seems like there's already no reason to use `typing_extensions` for anything.

(Runtime introspection features like `get_overloads()` weren't added to `typing` until Python 3.11, but Sphinx doesn't use those.)

Reply:
We don't use `get_overloads` for now but we might support it in the future (there is an open issue somewhere I think but I forgot where)